models/DataObject/ClassDefinition/Data/Multiselect.php

```diff
@@ -645,6 +645,35 @@ public function resolveBlockedVars(): array
         return $blockedVars;
     }
 
+    /**
+     * Perform a sanity check on option values, as a comma in option value leads to silent data corruption when persisted, see #5010.
+     *
+     * This method is called in DataObject\ClassDefinition::save()
+     *
+     * @param DataObject\ClassDefinition $class
+     * @param array $params
+     */
+    public function classSaved($class, $params = [])
+    {
+        $optionsProvider = DataObject\ClassDefinition\Helper\OptionsProviderResolver::resolveProvider(
+            $this->getOptionsProviderClass(),
+            DataObject\ClassDefinition\Helper\OptionsProviderResolver::MODE_MULTISELECT
+        );
+        if ($optionsProvider) {
+            $context = [];
+            $context['fieldname'] = $this->getName();
+
+            $options = $optionsProvider->{'getOptions'}($context, $this);
+        } else {
+            $options = $this->getOptions();
+        }
+        if (is_array($options) && array_reduce($options, static function ($containsComma, $option) {
+            return $containsComma || (strpos($option['value'], ',') !== false);
```

Comment:
Since Pimcore 10 uses PHP 8, we should use [str_contains](https://www.php.net/str_contains):

```suggestion
            return $containsComma || str_contains($option['value'], ',');
```